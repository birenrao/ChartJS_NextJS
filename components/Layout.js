import React from 'react';
import styles from '../styles/Layout.module.css';

const Layout = ({ children }) => (
  <div className={styles.container}>
    <header>
      <h1>Welcome to Data Visualisation</h1>
    </header>
    <main className={styles.main}>
      {children}
    </main>
    <footer>
      <p>This is intellisense.io</p>
    </footer>
  </div>
);

export default Layout;
