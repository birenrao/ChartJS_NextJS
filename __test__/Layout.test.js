import React from 'react'
import { render } from '@testing-library/react'
import ShallowRenderer from 'react-test-renderer/shallow';

import Layout from '../components/Layout'

describe("Layout function", () => {
    
    test('renders Header', () => {
        const { getByText } = render(<Layout />)
        const header = getByText('Welcome to Data Visualisation')
        expect(header).toBeInTheDocument()
    });

    test('renders Footer', () => {
        const { getByText } = render(<Layout />)
        const footer = getByText('This is intellisense.io')
        expect(footer).toBeInTheDocument()
    });
    it('renders Layout unchanged', () => {
        const renderer = new ShallowRenderer();
        renderer.render(<Layout />);
        const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot()
    })

  });
