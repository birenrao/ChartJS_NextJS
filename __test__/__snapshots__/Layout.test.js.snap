// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout function renders Layout unchanged 1`] = `
<div
  className="container"
>
  <header>
    <h1>
      Welcome to Data Visualisation
    </h1>
  </header>
  <main
    className="main"
  />
  <footer>
    <p>
      This is intellisense.io
    </p>
  </footer>
</div>
`;
